pkg/api/types.go

```diff
@@ -1205,6 +1221,13 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// Local represents a directory with specific topology contraints
+type LocalVolumeSource struct {
+	// The full path to the volume on the node
+	// For alpha, this path must be a file-based volume
+	Path string
```

Comment:
I guess the main issue today is that this is the first volume plugin that can export both a file or block volume.  All other plugins today are only block or only file.  So the options I see are:

1.  Having a file struct and a block struct within the same volume source object.  None of the other volume objects do this, so it looks inconsistent.
2.  Overload Path field and use PV.VolumeType field to distinguish between the two.  The Path field having different behaviors/semantics depending on the type may be difficult for users to understand.
3.  Have two completely separate volume plugins, one for local file volumes, and one for local block volumes.  This is more consistent with the current semantics (kinda like rbd and cephfs).  We are trying not to hardcode the Local volume type into the scheduler and pv controller, so adding a block volume plugin in the future should work as long as it uses the same general topology mechanisms.

What do you guys think about the 3rd approach?

On a side note, I think when we support local block devices, and we go with the new block access model, then we won't need fstype in the volume spec.  We'll just use the fstype in the PVC.